contracts/registry/NameRegistry.sol

```diff
@@ -0,0 +1,124 @@
+        _;
+    }
+
+    modifier whitelistedFactoryOnly() {
+        require(whitelistedFactories[msg.sender], "NameRegistry: only whitelisted factory can call this function");
+        _;
+    }
+
+    //-----------------------------
+    //  INameRegistry IMPL - WRITE
+    //-----------------------------
+    function transferOwnership(address newOwner) external override ownerOnly {
+        address oldOwner = owner;
+        owner = newOwner;
+        emit TransferOwnership(oldOwner, newOwner, block.timestamp);
+    }
+
+    function setFactories(address[] memory factories, bool[] memory active) external override ownerOnly {
+        _setFactories(factories, active);
+    }
+
+    function mapIssuer(string memory name, address instance) external override whitelistedFactoryOnly {
+        issuerNameToAddressMap[name] = instance;
+        issuerAddressToNameMap[instance] = name;
+        emit MapIssuer(msg.sender, name, instance, block.timestamp);
+    }
+
+    function mapAsset(string memory name, address instance) external override whitelistedFactoryOnly {
+        assetNameToAddressMap[name] = instance;
+        assetAddressToNameMap[instance] = name;
+        emit MapAsset(msg.sender, name, instance, block.timestamp);
+    }
+
+    function mapCampaign(string memory name, address instance) external override whitelistedFactoryOnly {
+        campaignNameToAddressMap[name] = instance;
+        campaignAddressToNameMap[instance] = name;
+        emit MapCampaign(msg.sender, name, instance, block.timestamp);
+    }
+
+    //-----------------------------
+    //  INameRegistry IMPL - READ
+    //-----------------------------
+    function getIssuer(string memory name) external override view returns (address) {
+        return issuerNameToAddressMap[name];
+    }
+
+    function getIssuerName(address issuer) external view override returns (string memory) {
+        return issuerAddressToNameMap[issuer];
+    }
+
+    function getAsset(string memory name) external override view returns (address) {
+        return assetNameToAddressMap[name];
+    }
+
+    function getAssetName(address asset) external view override returns (string memory) {
+        return assetAddressToNameMap[asset];
+    }
+
+    function getCampaign(string memory name) external override view returns (address) {
+        return campaignNameToAddressMap[name];
+    }
+
+    function getCampaignName(address campaign) external view override returns (string memory) {
+        return campaignAddressToNameMap[campaign];
+    }
+
+    //------------------------
+    //  Helpers
+    //------------------------
+    function _setFactories(address[] memory factories, bool[] memory isWhitelisted) private {
+        require(
+            factories.length == isWhitelisted.length,
+            "NameRegistry: factoryAddress and fectoryStatus array size mismatch"
+        );
+        for (uint256 i = 0; i < factories.length; i++) {
```

Comment:
Why not `uint8`?

Reply:
read only function so I think there's no use in optimising this